pkg/auth/server/login/login.go

```diff
@@ -133,19 +159,98 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 	l.auth.AuthenticationSucceeded(context, then, w, req)
 }
 
-var DefaultLoginFormRenderer = loginTemplateRenderer{}
+// NewLoginFormRenderer creates a login form renderer that takes in an optional custom template to
+// allow branding of the login page. Uses the default if customLoginTemplateFile is not set.
+func NewLoginFormRenderer(templates *configapi.OAuthTemplates) (*loginTemplateRenderer, error) {
+	r := &loginTemplateRenderer{}
+	if templates != nil && len(templates.Login) > 0 {
+		customTemplate, err := template.ParseFiles(templates.Login)
+		if err != nil {
+			return nil, err
+		}
+		r.loginTemplate = customTemplate
+	} else {
+		r.loginTemplate = defaultLoginTemplate
+	}
+
+	return r, nil
+}
+
+func ValidateLoginTemplate(templates *configapi.OAuthTemplates) fielderrors.ValidationErrorList {
```

Comment:
we also want a unit test that calls `ValidateLoginTemplate` with a snapshot of the example template we release with this, with scary comments around the test saying "This template must always remain valid to avoid breaking compatibility with templates in use by customers"